### ISSUE TITLE
get_essence_data not working for all APIs

### ISSUE THREAD

**Issue by sara-chronister** [CDCgov/Rnssp#9]
The `get_essence_data()` function does not work with all APIs. Below is an example, copied directly from ESSENCE with no modifications:

```{r}

url <- "https://essence2.syndromicsurveillance.org/nssp_essence/api/tableBuilder?datasource=va_hosp&startDate=1Jan2013&medicalGroupingSystem=essencesyndromes&userId=5099&endDate=9May2023&percentParam=noPercent&site=934&hospFacilityType=emergency%20care&aqtTarget=TableBuilder&geographySystem=hospital&detector=nodetectordetector&timeResolution=daily&hasBeenE=1&rowFields=timeResolution&columnField=site"

data <- Rnssp::get_essence_data(url)

```

returns error: Error in `Rnssp::get_essence_data()`:
! URL is not of ESSENCE type. Check your URL or use `get_api_data()` instead!

I do not want to use `get_api_data()` for a number of reasons, the big one being that I still have to manipulate it into a data frame and the point of `get_essence_data()` is to avoid the extra required steps every time.

When I run my version of `get_essence_data()` it works:

```{r}

function(url, start_date = NULL, end_date = NULL) {
  
  api_type <- str_extract(url,"(?<=api/).+(?=\\?)")
  
  url_new <- Rnssp::change_dates(url, start_date, end_date)
  
  if (api_type == "timeSeries") {
    api_response <- myProfile$get_api_response(url_new)
    api_response_json <- content(api_response, as = "text")
    api_data <- fromJSON(api_response_json) %>%
      extract2("timeSeriesData")
  } else if (api_type == "timeSeries/graph") {
    api_png <- myProfile$get_api_tsgraph(url_new)
    knitr::include_graphics(api_png$tsgraph)
  } else if (api_type == "tableBuilder") {
    api_data <- myProfile$get_api_data(url_new)
  } else if (api_type == "tableBuilder/csv") {
    api_data <- myProfile$get_api_data(url_new, fromCSV = TRUE)
  } else if (api_type == "dataDetails") {
    api_data <- myProfile$get_api_data(url_new) %>%
      extract2("dataDetails")
  } else if (api_type == "dataDetails/csv") {
    api_data <- myProfile$get_api_data(url_new, fromCSV = TRUE) 
  } else if (api_type == "summaryData") {
    api_data <- myProfile$get_api_data(url_new) %>%
      extract2("summaryData")
  } else if (api_type == "alerts/regionSyndromeAlerts") {
    api_data <- myProfile$get_api_data(url_new) %>%
      extract2("regionSyndromeAlerts")
  } else if (api_type == "alerts/hospitalSyndromeAlerts") {
    api_data <- myProfile$get_api_data(url_new) %>%
      extract2("hospitalSyndromeAlerts")
  } else {
    writeLines("Error: API did not work as expected. Please check your URL, dates, and password before trying again.")
  }
  
}

```

**Comment by rosericazondekon** [comment]
Thank you @sara-chronister for reporting this bug in the `get_essence_data()` function. A patch will be pushed shortly to the development version of the package. The fix will ultimately be part of the next version (`Rnssp v0.2.1`) of the package when we release it.
Again, thank you so much for your contribution.

**Comment by rosericazondekon** [comment]
Hi @sara-chronister , I pulled using the URL you provided with the `get_api_data()` function and it does not require to manipulate the api response into a data frame. 
Before the release of the upcoming patch version, the `get_api_data()` function may be used without the need to manipulate the response into a dataframe.

**Comment by rosericazondekon** [comment]
A fix has been pushed to the [dev branch](https://github.com/CDCgov/Rnssp/blob/dev/R/api_utilities.R). Feel free to check it. Again, this bug fix will be part of `Rnssp v0.2.1` when we release it.
